content/blog/getting-started-with-k8s-part3/index.md

```diff
@@ -0,0 +1,372 @@
+                            runAsUser: 10001,
+                            capabilities: {
+                                drop: ["all"],
+                                add: ["NET_BIND_SERVICE"],
+                            },
+                            readOnlyRootFilesystem: true,
+                        },
+                        volumeMounts: [
+                            {
+                                mountPath: "/tmp",
+                                name: "tmp-volume",
+                            },
+                        ],
+                    },
+                ],
+                volumes: [
+                    {
+                        name: "tmp-volume",
+                        emptyDir: {
+                            medium: "Memory",
+                        },
+                    },
+                ],
+                nodeSelector: {
+                    "beta.kubernetes.io/os": "linux",
+                },
+            },
+        },
+    },
+});
+
+const ordersService = new k8s.core.v1.Service("orders", {
+    metadata: {
+        name: "orders",
+        labels: {
+            name: "orders",
+        },
+        namespace: sockShopNs.metadata.name,
+    },
+    spec: {
+        ports: [
+            {
+                port: 80,
+                targetPort: 80,
+            },
+        ],
+        selector: orders.spec.template.metadata.labels,
+    },
+});
+```
+
+If we look at the [manifest](https://github.com/microservices-demo/microservices-demo/blob/master/deploy/kubernetes/manifests/orders-db-dep.yaml) for the orders-db deployment, we can see that the code follows it closely. You can find the full [Sock Shop code in the Pulumi examples repository](https://github.com/pulumi/examples/tree/master/kubernetes-ts-sock-shop).
+
+Although we encapsulate the deployment in a single large script, we can improve on that by creating components using classes to abstract the implementation details.
+
+## Deploying Guestbook as a Component
+
+The Kubernetes [GuestBook](https://kubernetes.io/docs/tutorials/stateless-application/guestbook/) is a canonical example of a stateless application. It consists of multiple PHP frontend services backed by Redis replicas and a lead Redis database to store guestbook entries. In the previous example, we created a Deployment and a Service for each microservice in the application. Although this is a valid pattern for deploying an application in Kubernetes, it doesn’t take full advantage of the capabilities of infrastructure as code.
+
+We can abstract objects, such as Deployments and Services, into components to create our own resources. [ComponentResource]({{< relref "/docs/reference/pkg/nodejs/pulumi/pulumi#ComponentResource" >}}) is Pulumi construct or resource that can aggregate other resources into a combined resource. Let’s take a look at how we implement the [Guestbook with components](https://github.com/pulumi/examples/tree/master/kubernetes-ts-guestbook/components).
```

Comment:
Consider moving this paragraph below the code snippet just below it, but before the `ComponentResource` example

Reply:
accepted